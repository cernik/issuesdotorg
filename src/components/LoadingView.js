// @flow
import React from 'react';
import {View, ActivityIndicator, StyleSheet} from 'react-native';

type LoadingViewProps = {
  fill?: boolean,
};

const LoadingView = ({fill = false}: LoadingViewProps) => (
  <View
    style={[
      styles.container,
      {
        flex: +fill,
      },
    ]}>
    <ActivityIndicator size="large" color="black" />
  </View>
);

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 32,
  },
});

export default LoadingView;
