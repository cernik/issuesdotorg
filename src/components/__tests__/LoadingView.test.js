// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import LoadingView from '../LoadingView';

describe('snapshot', () => {
  it('renders correctly with no data', () => {
    const tree = renderer.create(<LoadingView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
