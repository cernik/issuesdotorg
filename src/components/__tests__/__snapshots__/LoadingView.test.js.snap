// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly with no data 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "justifyContent": "center",
        "padding": 32,
      },
      Object {
        "flex": 0,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="black"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
